File: parent335/pinyougou-interface/src/main/java/cn/itcast/core/service/user/UserService.java
package cn.itcast.core.service.user;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.ad.Content;
import cn.itcast.core.pojo.order.Order;
import cn.itcast.core.pojo.user.User;
import com.alibaba.dubbo.config.annotation.Service;

import java.util.List;

@Service
public interface UserService {

    /**
     * 用户获取短信验证码
     * @param phone
     */
    public void sendCode(String phone);

    /**
     * 用户注册
     * @param user
     * @param smscode
     */
    public void add(User user, String smscode);

    /**
     * c
     *
     * @param id
     * @return
     */
    public User findById(Long id);
    public void upDate(User user);

    public List<User> findOneByuserName(String userName);
    public PageResult search(String userName, Integer page, Integer rows, Order order);

    /**
     * 查询
     * @param name
     * @return
     */
    public List<Long> findById2(String name);
//
}

File: parent335/pinyougou-web-user/src/main/webapp/js/service/userService.js
//服务层
app.service('userService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../usermanage/findAll.do');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../usermanage/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../usermanage/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity,smscode){
		return  $http.post('../usermanage/add.do?smscode='+smscode,entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../usermanage/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../usermanage/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../usermanage/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	//发送验证码
	this.sendCode=function(phone){
		return $http.get('../usermanage/sendCode.do?phone='+phone);
	}
	
});
